xlators/cluster/dht/src/dht-rebalance.c

```diff
@@ -3136,18 +3152,30 @@ gf_defrag_get_entry(xlator_t *this, int i, dht_container_t **container,
                                         conf->link_xattr_name);
 
         if (is_linkfile) {
-            /* No need to add linkto file to the queue for
-               migration. Only the actual data file need to
-               be checked for migration criteria.
-            */
-
-            gf_msg_debug(this->name, 0,
-                         "Skipping linkfile"
-                         " %s on subvol: %s",
-                         df_entry->d_name, conf->local_subvols[i]->name);
-            continue;
+            for (j = 0; j < conf->subvolume_cnt; j++) {
+                if (conf->decommissioned_bricks[j] == conf->local_subvols[i]) {
+                    loc_wipe(&entry_loc);
+                    ret = dht_build_child_loc(conf->local_subvols[i],
+                                              &entry_loc, loc,
+                                              df_entry->d_name);
+                    if (ret) {
+                        gf_log(this->name, GF_LOG_ERROR,
+                               "Child loc"
+                               " build failed for entry: %s",
+                               df_entry->d_name);
+                    } else {
+                        ret = syncop_unlink(conf->local_subvols[i], &entry_loc,
+                                            NULL, NULL);
+                    }
+                    gf_msg_debug(this->name, 0,
+                                 "Unlink linkfile"
+                                 " %s on subvol: %s status is %d",
+                                 df_entry->d_name, conf->local_subvols[i]->name,
+                                 ret);
+                    continue;
```

Comment:
This `continue` just continues the `for` loop at line 3155. It should be a `break`, and add the `continue` unconditionally after the `for` loop.

Reply:
We can't blindly continue for every linkfile, we need to skip only linkfiles that belong to decommissioned brick, rest of them we need to add in a queue to unlink by the function gf_defrag_migrate_single_file.